src/applications/health-care-questionnaire/questionnaire/components/hidden-fields/HiddenFields.jsx

```diff
@@ -0,0 +1,38 @@
+import React, { useState, useEffect } from 'react';
+import { connect } from 'react-redux';
+
+import { getCurrentAppointmentId } from '../../utils';
+
+const HiddenFields = props => {
+  const { onChange, context } = props;
+  const [questionnaireId] = useState(context.questionnaire?.id);
+  const [appointmentId] = useState(getCurrentAppointmentId(window));
+
+  useEffect(
+    () => {
+      let timeout;
+      if (questionnaireId || appointmentId) {
+        timeout = setTimeout(
```

Comment:
Is something weirding out in the UI where we need this timeout? If we can avoid timeouts, that would be best as they tend to cause flakiness